options_filter.go

```diff
@@ -6,21 +6,34 @@ import "strings"
 // Use of this source code is governed by a license
 // that can be found in the LICENSE file.
 
-// OPTIONSFilter is a filter function that inspects the Http Request for the OPTIONS method
+// OptionsFilter is used to create a Filter that implements proper headers for CORS OPTIONS
```

Comment:
Renaming the type is breaking the api of this package. IMO this must be avoided.
It was spelled in uppercase because it reflects the OPTIONS method.